lib/parsers/package-lock-parser.ts

```diff
@@ -119,7 +120,13 @@ export class PackageLockParser implements LockfileParser {
       } else if (/^file:/.test(dep.version)) {
         depTree.dependencies[dep.name] = createPkgTreeFromDep(dep);
       } else {
-        throw new OutOfSyncError(depName, 'npm');
+        // TODO: also check the package version
+        // for a stricter check
+        if (strictOutOfSync) {
+          throw new OutOfSyncError(depName, 'npm');
+        }
+        depTree.dependencies[dep.name] = createPkgTreeFromDep(dep);
+        depTree.dependencies[dep.name].missingLockFileEntry = true;
```

Comment:
Let's maybe move setting up `missingLockFileEntry` flag to `createPkgTreeFromDep` function?

`createPkgTreeFromDep(dep: Dep)` => `createPkgTreeFromDep(dep: Dep, missingLockFileEntry: boolean = false)`

Reply:
@miiila I just don't like this sequence: call some function (`createPkgTreeFromDep`) => patch something in the result of the function `missingLockFileEntry = true`. It is hard to read and possibly dangerous (`Cannot set property 'missingLockFileEntry' of undefined`).

Though, you maybe right.